source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -284,48 +298,63 @@ bool Filter::readProxyHeader(os_fd_t fd) {
         header_version_ = V2;
       } else if (memcmp(buf_, PROXY_PROTO_V1_SIGNATURE, PROXY_PROTO_V1_SIGNATURE_LEN)) {
         // It is not v2, and can't be v1, so no sense hanging around: it is invalid
-        throw EnvoyException("failed to read proxy protocol (exceed max v1 header len)");
+          ENVOY_LOG(debug, "failed to read proxy protocol (exceed max v1 header len)");
+          return false;
       }
     }
 
     if (header_version_ == V2) {
       const int ver_cmd = buf_[PROXY_PROTO_V2_SIGNATURE_LEN];
       if (((ver_cmd & 0xf0) >> 4) != PROXY_PROTO_V2_VERSION) {
-        throw EnvoyException("Unsupported V2 proxy protocol version");
+          ENVOY_LOG(debug, "Unsupported V2 proxy protocol version");
+          return false;
       }
       if (buf_off_ < PROXY_PROTO_V2_HEADER_LEN) {
         ssize_t exp = PROXY_PROTO_V2_HEADER_LEN - buf_off_;
         const Api::SysCallSizeResult read_result = os_syscalls.recv(fd, buf_ + buf_off_, exp, 0);
         if (read_result.rc_ != exp) {
-          throw EnvoyException("failed to read proxy protocol (remote closed)");
+            ENVOY_LOG(debug, "failed to read proxy protocol (remote closed)");
+            return false;
         }
         buf_off_ += read_result.rc_;
         nread -= read_result.rc_;
       }
-      ssize_t addr_len = lenV2Address(buf_);
+      ssize_t addr_len = lenV2Address(buf_).value_or(NO_VALUE_ASSIGNED);
```

Comment:
Why not test against `has_value()`? That way you can remove `NO_VALUE_ASSIGNED`

Reply:
Perfect!